scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1084,19 +1084,64 @@
                               :first-ten-jobs (print-str (->> jobs (take 10) (map :job/uuid) (map str)))}))
       (dissoc compute-cluster->jobs :no-acceptable-compute-cluster))))
 
+(defn decrement-scheduling-capacity-counter
+  "Decrement capacity for compute cluster in the counter."
+  [capacity-counter compute-cluster]
+  (update-in capacity-counter [compute-cluster] dec))
+
+(defn scheduling-capacity-constrained-job-distributor
+  "Create a job distributor function that only assigns jobs to compute clusters
+   with available scheduling capacity. Assignments are tracked and the given
+   counter is decremented."
+  [compute-cluster->scheduling-capacity autoscalable-jobs pool-name compute-clusters job->acceptable-compute-clusters-fn]
+  (tracing/with-span [s {:name "scheduler.distribute-jobs-to-compute-cluster"
+                         :tags {:pool pool-name :component tracing-component-tag}}]
+    (let [capacity-counter-atom (atom compute-cluster->scheduling-capacity)
```

Comment:
I'd be more consistent with this atom's name. It sort of came from nowhere and doesn't have the `->` hint that its a map.

E.g., `compute-cluster->available-scheduling-capacity-atom`. Also should probably use the non-`atom` suffixed version as the argument name to `decrement-scheduling-capacity-counter` 

I'd not name it `compute-cluster->scheduling-capacity-atom` because its semantically different than that.

Reply:
done